core/src/main/scala/io/finch/Endpoint.scala

```diff
@@ -1118,4 +1118,16 @@ object Endpoint {
    */
   def multipartAttributesNel[F[_]: Sync, A: DecodeEntity: ClassTag](name: String): Endpoint[F, NonEmptyList[A]] =
     new Attribute[F, NonEmptyList, A](name) with Attribute.NonEmpty[F, A] with Attribute.MultipleErrors[F, NonEmptyList, A]
+
+  /**
+   * Sequentially composes the given `endpoints` by using [[Endpoint!.coproduct]].
+   *
+   * The resulting endpoint will match if at least one of the provided endpoints matches.
+   * If the sequence of provided endpoints is empty, the empty endpoint is returned, which never matches.
+   *
+   * @see [[Endpoint!.coproduct]] for the exact composition semantics.
+   * @see [[Endpoint.empty]] for the semantics of the empty endpoint.
+   */
+  def coproduct[F[_], A](endpoints: Endpoint[F, A]*): Endpoint[F, A] =
```

Comment:
I think that's good to have but I'm not super sure about the name, simply b/c we already have `Endpoint.coproduct`. Is there anything else we can name this?

Reply:
We have `Endpoint.coproduct` as an instance method but this one is on the companion object. Granted, they don't do exactly the same but are very closely related. Think of it as having `concat` on collections both as an instance and companion object method (although we only have `++`, but very close).

Anyway if you suggest another name it's fine by me.

Reply:
How about `fold` or `reduce`?

Reply:
`fold` and `reduce` already have a specific meaning in Scala. I could suggest also `sum`, `oneOf` or `atLeastOneOf`.

Reply:
`coproductAll`?

Reply:
Semantically, I like `oneOf`. That's exactly what this combinator does 🤔

Reply:
So which one should it be? I called it `coproductAll` right now. Or it could be `coproductOf` as well.

Reply:
Not sure if it's already discussed, but have you considered refactoring the existing `coproduct` instance method to support variable argument? Or overload with
```
final def coproduct[B >: A](other: Endpoint[F, B], rest: Endpoint[F, B]*): Endpoint[F, B]
```

It should be backward compatible. Cheers!

Reply:
I don't like it that much. It would look like there is special significance to the first endpoint (the method reciever) when there really isn't one.